lib/temporaryAllowlist.js

```diff
@@ -0,0 +1,65 @@
+/** @module temporaryAllowlist */
+
+// The contentBlocking allowlist only disables tracker blocking for a website.
+const CONTENT_BLOCKING_ALLOWLIST_PRIORITY = 20000
+// The unprotectedTemporary allowlist disables all protections for a website.
+const UNPROTECTED_TEMPORARY_ALLOWLIST_PRIORITY = 1000000
+
+const {
+    generateDNRRule
+} = require('./utils')
+
+/**
+ * @typedef generateTemporaryAllowlistRulesResult
+ * @property {import('./utils').DNRRule} rule
+ * @property {object} matchDetails
+ */
+
+/**
+ * Generator to produce the declarativeNetRequest rules and corresponding match
+ * details for the unprotectedTemporary and contentBlocking sections of the
+ * extension configuration.
+ * @param {object} extensionConfiguration
+ *   The extension configuration.
+ * @return {Generator<generateTemporaryAllowlistRulesResult>}
+ */
+function* generateTemporaryAllowlistRules (
```

Comment:
Do we need a generator for this?

Reply:
I used a generator for consistency with the other code, but it probably doesn't matter much either way. If there was a huge number of allowlisted domains, we'd use less memory by avoiding creating a huge array before iterating through it, but that's unlikely.